y/metrics.go

```diff
@@ -25,67 +25,121 @@ var (
+	numIteratorsCreated *expvar.Int
+	// numPuts is number of puts -> Number of puts requests made
 	numPuts *expvar.Int
-	// numBlockedPuts is number of blocked puts
-	numBlockedPuts *expvar.Int
-	// numMemtableGets is number of memtable gets
+	// numMemtableGets is number of memtable gets -> Number of get requests made on memtable
 	numMemtableGets *expvar.Int
 	// numCompactionTables is the number of tables being compacted
 	numCompactionTables *expvar.Int
+	// Total writes by a user in bytes
+	numBytesWrittenUser *expvar.Int
 )
 
 // These variables are global and have cumulative values for all kv stores.
 func init() {
-	numReads = expvar.NewInt("badger_v3_disk_reads_total")
-	numWrites = expvar.NewInt("badger_v3_disk_writes_total")
-	numBytesRead = expvar.NewInt("badger_v3_read_bytes")
-	numBytesWritten = expvar.NewInt("badger_v3_written_bytes")
-	numLSMGets = expvar.NewMap("badger_v3_lsm_level_gets_total")
-	numLSMBloomHits = expvar.NewMap("badger_v3_lsm_bloom_hits_total")
-	numGets = expvar.NewInt("badger_v3_gets_total")
-	numPuts = expvar.NewInt("badger_v3_puts_total")
-	numBlockedPuts = expvar.NewInt("badger_v3_blocked_puts_total")
-	numMemtableGets = expvar.NewInt("badger_v3_memtable_gets_total")
-	lsmSize = expvar.NewMap("badger_v3_lsm_size_bytes")
-	vlogSize = expvar.NewMap("badger_v3_vlog_size_bytes")
-	pendingWrites = expvar.NewMap("badger_v3_pending_writes_total")
-	numCompactionTables = expvar.NewInt("badger_v3_compactions_current")
+	numReadsVlog = expvar.NewInt("badger_v4_disk_reads_total")
+	numWritesVlog = expvar.NewInt("badger_v4_disk_writes_total")
+	numBytesVlogWritten = expvar.NewInt("badger_v4_vlog_written_bytes")
+	numBytesReadVlog = expvar.NewInt("badger_v4_read_bytes_vlog")
+
+	numBytesReadLSM = expvar.NewInt("badger_v4_read_bytes_lsm")
+	numBytesLSMWritten = expvar.NewInt("badger_v4_lsm_written_bytes")
+	numBytesCompactionWritten = expvar.NewInt("badger_v4_compaction_written_bytes")
+
+	numLSMGets = expvar.NewMap("badger_v4_lsm_level_gets_total")
+	numLSMBloomHits = expvar.NewMap("badger_v4_lsm_bloom_hits_total")
+	numMemtableGets = expvar.NewInt("badger_v4_memtable_gets_total")
+
+	// User operations
+	numGets = expvar.NewInt("badger_v4_gets_total")
+	numPuts = expvar.NewInt("badger_v4_puts_total")
+	numBytesWrittenUser = expvar.NewInt("badger_v4_write_user")
+
+	// Required for Enabled
+	// Do we need this? We don't use this anyway
+	numGetsWithResults = expvar.NewInt("badger_v4_get_results")
+	numIteratorsCreated = expvar.NewInt("badger_v4_iterators")
+
+	// Sizes
+	lsmSize = expvar.NewMap("badger_v4_lsm_size_bytes")
+	vlogSize = expvar.NewMap("badger_v4_vlog_size_bytes")
+
+	pendingWrites = expvar.NewMap("badger_v4_pending_writes_total")
+	numCompactionTables = expvar.NewInt("badger_v4_compactions_current")
+}
+
+func NumIteratorsCreatedAdd(enabled bool, val int64) {
+	addInt(enabled, numIteratorsCreated, val)
+}
+
+func NumGetsWithResultsAdd(enabled bool, val int64) {
```

Comment:
i would rather track the misses. if you are looking for latencies when results are fetched that should be handled separately